sktime/benchmarking/_lib_mini_kotsu/run.py

```diff
@@ -58,16 +65,37 @@ def run(
         results_df = pd.DataFrame(columns=["validation_id", "model_id", "runtime_secs"])
         results_df["runtime_secs"] = results_df["runtime_secs"].astype(int)
 
+    tqdm_available, _ = _check_soft_dependencies("tqdm", severity="warning")
```

Comment:
I think the warning should be raised only if `verbose`, with a clear error message. Otherwise, I would just leave it silent.

Reply:
If you want to warn, you should use `warn` from `sktime.utils.warnings` which receives a reference from `self`, so users can turn it off via `set_config`.

Reply:
@fkiraly 
```{python}
     try:
        results_df = pd.read_csv(results_path)
    except FileNotFoundError:
        results_df = pd.DataFrame(columns=["validation_id", "model_id", "runtime_secs"])
        results_df["runtime_secs"] = results_df["runtime_secs"].astype(int)

    tqdm_available, _ = _check_soft_dependencies("tqdm", severity="none")

    if verbose and not tqdm_available:
        warn("tqdm is not installed. Continuing without progress bars.", logger)

    if tqdm_available:
        from tqdm import tqdm
```
Something like this ⬆ ?

**This has not been committed!**

Reply:
yes, though use `from sktime.utils.warnings import warn`, there you can pass the `self` reference.